### ISSUE TITLE
InboxActor: EnqueueQuery and other methods should be private. Rewrite Receive

### ISSUE THREAD

**Issue by HCanber** [akkadotnet/akka.net#265]
### Public to private

`InboxActor` has many public methods (although the class is internal) that should not be called externally (and they are not currently).
For example, if `EnqueueQuery` was to be called externally it can introduce race conditions (`_clients` is updated but not `_clientsByTimeout`).
`MessagePredicate` and `EnqueueMessage` have concurrent problems as well.
Make all public methods private.
### Receive

Core actors should not be using `PatternMatch` (or `ReceiveActor`) due to performance and GC overhead. Rewrite `Receive` to raw C# by using `is` or `as`:

``` c#
var getMessage=message as Get;
if(getMessage != null)
{
   ...
}
```

**Comment by nbparrell** [comment]
I will be working on this.

**Comment by nbparrell** [comment]
@HCanber Just want to clarify, where you are asking for the rewrite of receive. Is that in the InboxActor or in the receiveActor that you are asking for the change?

**Comment by Aaronontheweb** [comment]
@nbparrell `InboxActor`

**Comment by Aaronontheweb** [comment]
nice @nbparrell ! Send in a PR - easier to comment there than on an issue.

**Comment by nbparrell** [comment]
where is the PR...... lol
Alittle new to the contributing scene.

**Comment by Aaronontheweb** [comment]
@nbparrell all good!

https://help.github.com/articles/about-pull-requests/ - this explains how to send in a pull request along with a little walkthrough. That should help!

**Comment by nagytech** [comment]
Ready for review in #4073

**Comment by Aaronontheweb** [comment]
closed via #4073
